### PR TITLE
refactor: redesign add module page

### PR DESCRIPTION
**Pull request by crowlKats**

### CODE REVIEW

File: islands/AddModule.tsx [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+                    Enter the above URL in the{" "}
+                    <span class={tw`font-semibold`}>payload URL</span> field.
+                  </li>
+                  <li>
+                    Select{" "}
+                    <span class={tw`font-semibold`}>application/json</span>{" "}
+                    as the content type.
+                  </li>
+                  <li>
+                    Select<span class={tw`font-semibold`}>
```
Comment (bartlomieju) [inline islands/AddModule.tsx:249]:
```suggestion
                    Select{" "}<span class={tw`font-semibold`}>
```

**Comment by bartlomieju** [comment]
This error is visible immediately after visiting the page

Screenshot 2022-07-20 at 21 50 04

**Comment by kitsonk** [comment]
Sorry for the late feedback, but it feels odd that this is copyable/visible until there is a valid package name:

Third_Party_Modules___Deno


Also, if you type a valid package name (e.g. `oak_se`) and then you type a taken valid name (e.g. `oak_sessions`) the confetti goes off and the UI updates for success:

Third_Party_Modules___Deno

**Comment by crowlKats** [comment]
@kitsonk fixed, PTAL

**Comment by kitsonk** [comment]
Maybe I misunderstood.  I am still getting "success" and confetti when going from a valid name to an invalid name:

Third_Party_Modules___Deno

**Comment by crowlKats** [comment]
huh, i cannot reproduce

**Comment by kitsonk** [comment]
https://user-images.githubusercontent.com/1282577/180760202-1ca3c74d-288b-4c6d-aca3-1611776e50d3.mp4

**Comment by crowlKats** [comment]
ah, after spamming delete and writing name fast, i could reproduce again...

**Comment by kitsonk** [comment]
In my video I wouldn't consider it fast typing. 😜

**Comment by crowlKats** [comment]
ok i really cannot reproduce the way you are doing it...

**Comment by roj1512** [comment]
@crowlKats you can try this with the name `twitter`, too

**Comment by crowlKats** [comment]
ok, so it seems you need to be typing at a specific speed and then it happens; love it.

**Comment by kitsonk** [comment]
> ok i really cannot reproduce the way you are doing it...

"works on my machine"... 🤣 Version 1.40.113 Chromium: 103.0.5060.114 (Official Build) (x86_64)

**Comment by kitsonk** [comment]
On Firefox 103, I have to type really fast, but it breaks in a different way, no "error" that module is taken and sort of infinite confetti that looks really yucky:

Third_Party_Modules___Deno_and_refactor__redesign_add_module_page_by_crowlKats_·_Pull_Request__2252_·_denoland_dotland

**Comment by crowlKats** [comment]
this is extremely annoying... how about now?

**Comment by kitsonk** [comment]
Same behaviour for me on: https://dotland-2m9t6qw1bv5g.deno.dev/add_module

**Comment by crowlKats** [comment]
@kitsonk check again

**Comment by kitsonk** [comment]
Ok, the confetti has gone and works as expected, thought my manual fuzzing has demonstrated another minor issue.  Long package names cause layout problems:

Third_Party_Modules___Deno

**Review by kitsonk — APPROVED** [review]
LGTM, though one minor nit about long module names and layout, but could be treated as a different issue.

**Comment by crowlKats** [comment]
oh, thats fine, thats simple to fix, thank god the false-positive is fixed now

**Pull request merged**